src/query/storages/parquet/src/parquet_rs/column_nodes.rs

```diff
@@ -0,0 +1,181 @@
+                let mut child_leaf_ids = Vec::with_capacity(inner_fields.len());
+                for inner_field in inner_fields.iter() {
+                    let child_column_node = Self::traverse_fields_dfs(inner_field, true, leaf_id);
+                    child_leaf_ids.extend(child_column_node.leaf_indices.clone());
+                    child_column_nodes.push(child_column_node);
+                }
+                ColumnNodeRS::new(
+                    field.clone(),
+                    true,
+                    child_leaf_ids,
+                    Some(child_column_nodes),
+                )
+            }
+            ArrowType::List(inner_field)
+            | ArrowType::LargeList(inner_field)
+            | ArrowType::FixedSizeList(inner_field, _) => {
+                let mut child_column_nodes = Vec::with_capacity(1);
+                let mut child_leaf_ids = Vec::with_capacity(1);
+                let child_column_node = Self::traverse_fields_dfs(inner_field, true, leaf_id);
+                child_leaf_ids.extend(child_column_node.leaf_indices.clone());
+                child_column_nodes.push(child_column_node);
+                ColumnNodeRS::new(
+                    field.clone(),
+                    true,
+                    child_leaf_ids,
+                    Some(child_column_nodes),
+                )
+            }
+            ArrowType::Map(inner_field, _) => {
+                let mut child_column_nodes = Vec::with_capacity(1);
+                let mut child_leaf_ids = Vec::with_capacity(1);
+                let child_column_node = Self::traverse_fields_dfs(inner_field, true, leaf_id);
+                child_leaf_ids.extend(child_column_node.leaf_indices.clone());
+                child_column_nodes.push(child_column_node);
+                ColumnNodeRS::new(
+                    field.clone(),
+                    true,
+                    child_leaf_ids,
+                    Some(child_column_nodes),
+                )
+            }
+            _ => {
+                let column_node = ColumnNodeRS::new(field.clone(), is_nested, vec![*leaf_id], None);
+                *leaf_id += 1;
+                column_node
+            }
+        }
+    }
+
+    pub fn traverse_path<'a>(
+        column_nodes: &'a [ColumnNodeRS],
+        path: &'a [usize],
+    ) -> Result<&'a ColumnNodeRS> {
+        let column_node = &column_nodes[path[0]];
+        if path.len() > 1 {
+            return match &column_node.children {
+                Some(ref children) => Self::traverse_path(children, &path[1..]),
+                None => Err(ErrorCode::Internal(format!(
+                    "Cannot get column_node by path: {:?}",
+                    path
+                ))),
+            };
+        }
+        Ok(column_node)
+    }
+}
+
+/// `ColumnNode` contains all the leaf column ids of the column.
+/// For the nested types, it may contain more than one leaf column.
+#[derive(Debug, Clone)]
+pub struct ColumnNodeRS {
```

Comment:
How about moving `ColumnNodeRS` and `convert` related functions to `src/common/storage/src`? This will make it easier to reuse the code in the future.

Reply:
@b41sh all fixed except this one, I  dislike the `RS` suffix so not want to pub it,  and I guess we are will not use arrow-rs for fuse in a near future.  I may do small refactors to these code, keep them in a a crate is more convinient.

Reply:
ok